emcc.py

```diff
@@ -2605,6 +2605,8 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
     if not shared.Settings.WASM_BACKEND and not DEBUG:
       os.unlink(asm_target) # we don't need the asm.js, it can just confuse
     sys.exit(0) # and we are done.
+  else:
+    shared.WebAssembly.add_emscripten_metadata(final, wasm_binary_target)
```

Comment:
looks like this is in the else of `SIDE_MODULE`? why not emit this in that case too?

Reply:
That's a good question. My reasoning is that side modules seem more like pure WASM modules and aren't really as constrained as main modules. Thinking about it a bit more, this is probably not true. All side modules will have to conform to the same ABI as the main module.

Reply:
We didn't give any guarantees there - that is, people should build them with the exact same version to be sure they'd work, since we didn't fix a final ABI yet. So sounds like the same situation, and we may as well emit it in all cases I think.